pkg/scheduler/objects/node_collection_test.go

```diff
@@ -120,3 +121,56 @@ func TestNodeCollection_GetNodes(t *testing.T) {
 	nodes = nc.GetNodes()
 	assert.Equal(t, 1, len(nodes), "list is missing node")
 }
+
+func TestGetNodeSortingPolicy(t *testing.T) {
+	nc := NewNodeCollection("test")
+	weights := map[string]float64{
+		"vcore":  2.0,
+		"memory": 3.0,
+	}
+
+	var tests = []struct {
+		input  string
+		except string
+	}{
+		{"fair", "fair"},
+		{"bin", "bin"},
+	}
+
+	for _, tt := range tests {
+		testname := fmt.Sprintf("Get method of base_collection:%s %s", tt.input, tt.except)
+		t.Run(testname, func(t *testing.T) {
+			policy := NewNodeSortingPolicy(tt.input, weights)
+			nc.SetNodeSortingPolicy(policy)
```

Comment:
Changing the policy changes the node ordering. Should this not be part of the test?
* nil sorting returns node in ??random?? order (not sure)
* fair sorting will return least loaded node first
* bin sorting should return most loaded node first.

We should have thus at least two nodes and make sure that the order they come back in is correct after the update.

Reply:
Got it.
I will add these scenarios in SetSortingPolicy and GetNodeIterator unit test.
Thanks for your feedback, @wilfred-s